### PR TITLE
Eslint-Prettier integration & pre commit hook

### PR DESCRIPTION
**Pull request by vnishukov**
### Motivation and context
Why:
https://prettier.io/docs/en/comparison.html

Description:
Using auto formatting tool like this integrated with project linter and pre-commit hooks is the fastest way to inline code styling for whole project

What it can do:
- Format code to look pretty well for reading and understanding
- Automatically add our header under commit process
- Automatically format weird writing code under commit process
- Automatically fix all autofixable stuffs using eslint under commit process
- Prevent commit if any linter error is present and it cant be autofixed. Write user what exactly the error is

Benefits:
- We will get unified code in the repo no matter how it was initially wrote
- We can be sure that all of ours linter rules will be followed each contributor even if he/she not completely know this rules
- Nice, clear and readable, pretty well formatted code which is easy to understand and support

---
See the auto formatted file as example below (shortcuts-dialog.tsx)

### How has this been tested?
Make a changes for unformatted file, make couple of linter issues and try to commit.
In case all issues are autofixable hook will format all stuffs automatically otherwise it will write the error to console

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@vnishukov , could you please fix merge conflicts?

**Comment by nmanovic** [comment]
@vnishukov , I don't see git hooks in the commit.

File: cvat-ui/package.json [hunk 1]
```diff
@@ -1,82 +1,99 @@
...
+        "react-redux": "^7.1.1",
+        "react-router": "^5.1.0",
+        "react-router-dom": "^5.1.0",
+        "react-share": "^3.0.1",
+        "redux": "^4.0.4",
+        "redux-devtools-extension": "^2.13.8",
+        "redux-logger": "^3.0.6",
+        "redux-thunk": "^2.3.0"
+    },
+    "husky": {
```
Comment (nmanovic) [inline cvat-ui/package.json:87]: OK. I see the line now.

**Comment by vnishukov** [comment]
> @vnishukov , could you please fix merge conflicts?

Sure, I will

**Review by bsekachev — APPROVED** [review]

**Pull request merged**